test/foundry-tests/TickBitmap.t.sol

```diff
@@ -0,0 +1,342 @@
+    using TickBitmap for mapping(int16 => uint256);
+
+    mapping(int16 => uint256) public bitmap;
+
+    function isInitialized(int24 tick) internal view returns (bool) {
+        (int24 next, bool initialized) = bitmap.nextInitializedTickWithinOneWord(tick, 1, true);
+        return next == tick ? initialized : false;
+    }
+
+    function flipTick(int24 tick) internal {
+        bitmap.flipTick(tick, 1);
+    }
+
+    function test_isInitialized_isFalseAtFirst() public {
+        assertEq(isInitialized(1), false);
+    }
+
+    function test_isInitialized_isFlippedByFlipTick() public {
+        flipTick(1);
+
+        assertEq(isInitialized(1), true);
+    }
+
+    function test_isInitialized_isFlippedBackByFlipTick() public {
+        flipTick(1);
+        flipTick(1);
+
+        assertEq(isInitialized(1), false);
+    }
+
+    function test_isInitialized_isNotChangedByAnotherFlipToADifferentTick() public {
+        flipTick(2);
+
+        assertEq(isInitialized(1), false);
+    }
+
+    function test_isInitialized_isNotChangedByAnotherFlipToADifferentTickOnAnotherWord() public {
+        flipTick(1 + 256);
+
+        assertEq(isInitialized(257), true);
+        assertEq(isInitialized(1), false);
+    }
+
+    function test_flipTick_flipsOnlyTheSpecifiedTick() public {
+        flipTick(-230);
+
+        assertEq(isInitialized(-230), true);
+        assertEq(isInitialized(-231), false);
+        assertEq(isInitialized(-229), false);
+        assertEq(isInitialized(-230 + 256), false);
+        assertEq(isInitialized(-230 - 256), false);
+
+        flipTick(-230);
+        assertEq(isInitialized(-230), false);
+        assertEq(isInitialized(-231), false);
+        assertEq(isInitialized(-229), false);
+        assertEq(isInitialized(-230 + 256), false);
+        assertEq(isInitialized(-230 - 256), false);
+
+        assertEq(isInitialized(1), false);
+    }
+
+    function test_flipTick_revertsOnlyItself() public {
+        flipTick(-230);
+        flipTick(-259);
+        flipTick(-229);
+        flipTick(500);
+        flipTick(-259);
+        flipTick(-229);
+        flipTick(-259);
+
+        assertEq(isInitialized(-259), true);
+        assertEq(isInitialized(-229), false);
+    }
+
+    function test_flipTick_gasCostOfFlippingFirstTickInWordToInitialized() public {
+        snapStart("flipTick_gasCostOfFlippingFirstTickInWordToInitialized");
+        flipTick(1);
+        snapEnd();
+    }
+
+    function test_flipTick_gasCostOfFlippingSecondTickInWordToInitialized() public {
+        flipTick(0);
+
+        snapStart("flipTick_gasCostOfFlippingSecondTickInWordToInitialized");
+        flipTick(1);
+        snapEnd();
+    }
+
+    function test_flipTick_gasCostOfFlippingATickThatResultsInDeletingAWord() public {
+        flipTick(0);
+
+        snapStart("flipTick_gasCostOfFlippingATickThatResultsInDeletingAWord");
+        flipTick(0);
+        snapEnd();
+    }
+
+    function setUpSomeTicks() internal {
```

Comment:
maybe stick this function towards the bottom of the test file in keeping with organization structure of other forge tests